packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_counter/blocs/counter_bloc.dart

```diff
@@ -37,37 +43,54 @@ abstract class CounterBlocStates {
 
   /// Error messages
   Stream<String> get errors;
+
+  /// Result state
+  Stream<Result<Count>> get counterResult;
 }
 
 /// A BloC responsible for count calculations
 @RxBloc()
 class CounterBloc extends $CounterBloc {
-  /// Default constructor
-  CounterBloc(this._repository);
+  /// Bloc constructor
+  CounterBloc(this._repository) {
+    Rx.merge<Result<Count>>([
+      // On increment.
+      _$incrementEvent
+          .switchMap((_) => _repository.increment().asResultStream()),
+      // On decrement.
+      _$decrementEvent
+          .switchMap((_) => _repository.decrement().asResultStream()),
+      // Get current value
+      _$currentEvent
+          .switchMap((_) => _repository.getCurrent().asResultStream()),
+      // Get initial value
+      _repository.getCurrent().asResultStream(),
+    ]).bind(_lastFetchedCount).disposedBy(_compositeSubscription);
+  }
 
   final CounterRepository _repository;
+  final _lastFetchedCount = BehaviorSubject<Result<Count>>();
 
-  /// Map increment and decrement events to `count` state
   @override
-  Stream<int> _mapToCountState() => Rx.merge<Result<Count>>([
-    // On increment.
-    _$incrementEvent
-        .switchMap((_) => _repository.increment().asResultStream()),
-    // On decrement.
-    _$decrementEvent
-        .switchMap((_) => _repository.decrement().asResultStream()),
-    // Get initial value
-    _repository.getCurrent().asResultStream(),
-  ])
-  // This automatically handles the error and loading state.
-      .setResultStateHandler(this)
-  // Provide success response only.
-      .whereSuccess()
-      .map((count) => count.value);
+  Stream<int> _mapToCountState() =>
+      _lastFetchedCount.whereSuccess().map((count) => count.value);
 
   @override
-  Stream<String> _mapToErrorsState() => errorState.toMessage();
+  Stream<String> _mapToErrorsState() => errorState.mapFromDio().toMessage();
 
   @override
   Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  Stream<Result<Count>> _mapToCounterResultState() =>
+      _lastFetchedCount.where((event) =>
+      event != Result.error(Exception()) ||
+          (event == Result.error(Exception()) &&
+              (event as DioError).response == null));
```

Comment:
Why do we need to filter out the result based on errors and exceptions?

Reply:
I think that this should be an old version of the code.

Reply:
Agreed. Don't know why it was suggested for review.